packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -1153,21 +1205,45 @@ class Auth {
      * A wrapper method for commerce-sdk-isomorphic helper: authorizeIDP.
      *
      */
-    async authorizeIDP(parameters: AuthorizeIDPParams) {
-        const redirectURI = parameters.redirectURI || this.redirectURI
+    async authorizeIDP(parameters: AuthorizeIDPPublicParams | AuthorizeIDPParams) {
+        const codeVerifier = helpers.createCodeVerifier()
+        const codeChallenge = await helpers.generateCodeChallenge(codeVerifier)
+        const organizationId = this.client.clientConfig.parameters.organizationId
+        const clientId = this.client.clientConfig.parameters.clientId
+        const siteId = this.client.clientConfig.parameters.siteId
         const usid = this.get('usid')
-        const customParameters = extractCustomParameters(parameters)
-        const {url, codeVerifier} = await helpers.authorizeIDP(
-            this.client,
-            {
-                redirectURI,
-                hint: parameters.hint,
-                ...(usid && {usid}),
-                ...customParameters
-            },
-            this.isPrivate
-        )
+        const dntPref = this.getDnt({includeDefaults: true})
+
+        // Handle both public and internal calls
+        const isPublicCall = 'redirectURI' in parameters && !('parameters' in parameters)
+        const redirectURI = isPublicCall
+            ? parameters.redirectURI
+            : parameters.parameters?.redirectURI || this.redirectURI
+        const hint = isPublicCall ? parameters.hint : parameters.parameters?.hint || ''
+        const customParams = isPublicCall ? parameters : parameters.parameters || {}
+
+        const url = `${
+            this.client.clientConfig.proxy || ''
+        }/shopper/auth/v1/organizations/${organizationId}/oauth2/authorize?${[
```

Comment:
Why is there a hard coded uri path to `/authorize` here? 

From the comment on this function, it should be calling helpers.authorizeIDP to generate the URL in question, no?

Reply:
agreed, this seems out of the place and duplicated logic from isormorphic

Reply:
I'm not sure why Joel originally built the URL manually, maybe it was because the isomorphic was not yet completed or released to NPM. 

I've updated the code to use `helpers.authorizeIDP` as suggested